docs/windows.md

```diff
@@ -0,0 +1,74 @@
+  associated filters must be included in the packages directory.
+- Windows has no make utility, so it must be brought in.
+- Windows does not have native support for bash, but does for powershell.
+- Windows does not have native support for utilities like curl, wget, grep, awk, sed and the such, but does for near-equivelent powershell cmdlets which can fullfill the same function. 
+- Windows does not support some os specific python modules, but has support for most others. 
+- There are differences between windows and linux/unix handle semantics and ioctls.
+- There are differences in file system semantics and file system conventions (i.e, "/opt/mesosphere" vs "c:/Mesosphere")
+- Windows does not support systemd or journald.
+- Windows docker operation is very different from linux
+- Many of the packages used in the linux deployment are different in the windows environment.
+- Windows machines usually prefer pre-built packages rather than in-situ builds from source.
+
+## Design Goals
+
+The design goals of goals of the Windows deployment are:
+- Make differences as consistent and predictable as possible. In principle a linux DC/OS system administrator should be able to learn a few transformation rules between the two environments and predict how to install, modify and operate a Windows host using the same skills they have already developed in linux.
+- Make the deployment also Windows native. A Windows sysadmin should be able to apply their knowlege hand habits as much as possible.
+- Make the deployment functionally equivalent (so far as possible) to the linux deployment.  A DC/OS UI user should not be able to obviously tell the difference between a linux and windows host without looking at the host attributes.
+
+The installation adds to those goals:
+- The deployment should be feasible on a Windows host without additional preparation. As a result, the deployment must use only tools available on a freshly installed windows machine.
+
+## changes to the dcos/dcos repository
+
+The package tree, top level scripts, and configuration files are, as much as possible, organised in a parallel manner to the
+linux build process, but they are not the same, so shell scripts are shadowed by powershell scripts of the same name, ie build_local becomes build_local.ps1, prep_local becomes prep_local.ps1 and so on.  
+
+- The package tree is paralleled by packages.windows. 
+  - In each directory we have a buildinfo.json parallel to the linux buildinfo.json, but usually different as windows requires different dependencies, packages, git repositories or branches, and commit shas from the linux equivelant. 
+  - The build script located in each of the pacakge directories is paralleled by a build.ps1 script which is functionally similar.
+  - In the packages.windows tree there are setupinfo.json and setup.ps1 scripts wwhich provide setup information at package 
+    install time. Those files get included in the tar.xz build artifact.
```

Comment:
The `packages.windows/` directory duplicates a lot of buildinfos from `packages/`. This adds to maintenance burden (more places a contributor needs to remember to update when modifying a package) and complexity (a contributor forgot to update the Windows version package, now we have drift between the two versions). Instead we should try to reuse buildinfos rather than copy them. This would simplify the act of updating a package and give us confidence that all versions of it have been built from the same source when that's the intent. This also keeps Windows and *nix build scripts that are meant to be doing the same thing far from each other, which increases the likelihood of drift between them. Basically I'd like to keep all of these Windows buildinfos and build scripts in the `packages/` directory, and I think we can do that mostly using existing Pkgpanda features.

Pkgpanda has support for what it calls "build variants". These are variants of a release that may contain a different set of packages, and some of those packages may be "variant" versions of the original. We should explore using that feature instead of creating a whole new package dir just for Windows. We can use package variants for cases were we want to swap in a "Windows variant" package that's meant to serve the same purpose but is built from difference sources, e.g. adminrouter in this PR. We can create a `packages/windows.treeinfo.json` that pulls in the `windows` variant of adminrouter, which would be defined at `packages/adminrouter/windows.buildinfo.json`. The `windows` build variant that results from building `windows.treeinfo.json` would then contain that package variant. We still have the different package, but now all the adminrouter-related buildinfos live together at `packages/adminrouter/`.

I'd also like to explore making these `build.ps1` scripts live alongside their `build` counterparts in the original packages (maybe with a more descriptive name, like `build.windows.ps1` or something). Keeping these files next to each other goes a long way to communicate that they're related, and makes it more likely that they'll stay in sync. And that way we can reuse the existing `buildinfo.json` without copying it, in the cases where it doesn't need to change for Windows.

So to summarize: we could define a `windows` build variant by adding a file at `packages/windows.treeinfo.json` which pulls in `windows` package variants that are defined at `packages/<package-name>/windows.buildinfo.json`. And then we could add `build.windows.ps1` scripts for all packages that need them in their existing directories, and teach Pkgpanda to use that script instead of `build` when it's running on Windows (looks like that much is already implemented). We'd then need to teach release tooling to know whether to build the `windows` variant, but that should be relatively simple: we'd omit building the `windows` treeinfo if we're not running on Windows.

Reply:
I will investigate this on Tuesday. I may need a conference call with you to better understand how to make this actually happen.

Reply:
Another thought is to go with what we have for the initial work, allowing us to start getting CI set up and more easily move forwards with less chance of breaking Linux, but create an issue for changing this as we progress further through the porting process

Reply:
Fixed.